pkg/scaler/utils.go

```diff
@@ -57,3 +59,25 @@ func isRateLimited(err error) bool {
 	const prefix = "rate: "
 	return strings.HasPrefix(err.Error(), prefix)
 }
+
+func shootWokeUp(old, new *gardenerv1alpha1.Cluster) bool {
+	decoder, err := extensionscontroller.NewGardenDecoder()
+	if err != nil {
+		return false
+	}
+
+	oldShoot, err := extensionscontroller.ShootFromCluster(decoder, old)
+	if err != nil {
+		return false
+	}
+	newShoot, err := extensionscontroller.ShootFromCluster(decoder, new)
+	if err != nil {
+		return false
+	}
+
+	if oldShoot.Status.IsHibernated == true && newShoot.Status.IsHibernated == false {
```

Comment:
Shouldn't we check the spec also here?

Reply:
This function is only to check the type of `update` event on the `cluster`. Returns true only if the `status` changed from "sleeping/waking up" to "awake/ready". If the cluster is in any other transition, then it would result false at which point, we wouldn't want to "enqueue" the cluster key. wdyt?